sections/implementation/api-structure.tex

```diff
@@ -1,17 +1,197 @@
+The first function \texttt{listAll} takes a request and a response as arguments and returns a \texttt{Promise<Response>}.
+Each function in the controllers is tagged with \texttt{async} to allow them to be called asynchronously. 
+\texttt{listAll} gets all the users from the database using the \texttt{userService.getAll()}.
+A response is then returned with the status code 200, meaning that request has succeded, and the users are returned in the response. 
+\\\\
+The next function \texttt{getOneById} is an endpoint used to get a user with a specific id. 
+Here we get the user id from the \texttt{req} object. 
+The id is found in the end of the URL for the endpoint, just as shown in \autoref{endpoint-example}.
+If we fail to find a user with the given id, we log the error and return a response with the status code 404, meaning that the resource was not found.
+If we find the user, we return the user object in the response with status code 200, meaning that the request was successful.
+\\\\
+As seen in the code examples for \texttt{userController}, we are calling functions from \texttt{userService}. 
+For entity we have, we have also created a service responsible for extracting and inserting data from the database.
+This is because they contain functionality that is needed in multiple endpoints, so the services are created to avoid duplicated code.
+\begin{lstlisting}[caption={Shows the function used to get get all ratings from the rating service}, captionpos=b, label={rating-service}]
+    class RatingService {
+        static getAll = async (): Promise<Rating[]> => {
+            //Get all ratings from database
+            const ratingRepository: Repository<Rating> = getRepository(Rating);
+            const ratings: Rating[] = await ratingRepository
+                .createQueryBuilder('rating')
+                .select(['rating.rating'])
+                .innerJoin('rating.user', 'user')
+                .innerJoin('rating.service', 'service')
+                .addSelect(['user.id', 'service.id'])
+                .getMany();
+    
+            return ratings;
+        };
+    }
+\end{lstlisting}
+\autoref{rating-service} shows the function \texttt{getAll} from the \texttt{RatingService}.
+This function extracts all ratings stored in the database and returns them as \texttt{Rating} objects in an array.
+First of all the \texttt{ratingRepository} is created, making it possible to create queries to the rating table in the database. 
+\\\\
+The \texttt{createQueryBuilder('rating')} starts we query to the rating table. 
+We then select the column \texttt{rating} from the rating table. 
+With \texttt{innerJoin()} we can join tables
+Because we have defined a many-to-one relationship on the \texttt{rating} entity class, we can join rating and user by referencing the \texttt{user} attribute on the \texttt{rating} entity and then specify that it should be joined with the \texttt{user} in the \texttt{innerJoin()} function. 
```

Comment:
```suggestion
Because we have defined a many-to-one relationship on the \texttt{rating} entity class, we can join rating and user by referencing the \texttt{user} attribute on the \texttt{rating} entity, and then specify that it should be joined with the \texttt{user} in the \texttt{innerJoin()} function. 
```